flycheck.el

```diff
@@ -9329,6 +9329,8 @@ This syntax checker requires Rust 1.15 or newer.  See URL
   :enabled (lambda ()
              (-when-let (file (buffer-file-name))
```

Comment:
Just noticing this now: I think it'd be simpler to use `buffer-file-name` (the variable) instead of the function.  As in:

```
(when buffer-file-name 
  (locate-dominating-file buffer-file-name "Cargo.toml"))
```
  
Or even this:

```
(and buffer-file-name 
     (locate-dominating-file buffer-file-name "Cargo.toml"))
```

Reply:
Should we change to the variable elsewhere in the code?  There are 80 occurrences.  Not only in checkers, but in the core code as well.

Reply:
I wonder.  I'm not sure why there's a function and a variable, except for the convenience of passing the function around.

I'm all for dropping things like `(when-let (fname (buffer-file-name)) …)`; I don't think it's worth polluting the history in cases where it's just cosmetic.

Reply:
I found a few places with this pattern, but I don't think using the variable would improve the situation:

In `flycheck-haskell--find-default-directory`:
```
     (or
      (when (buffer-file-name)
        (flycheck--locate-dominating-file-matching
         (file-name-directory (buffer-file-name))
         "stack.*\\.yaml\\'"))
      (-when-let* ((stack (funcall flycheck-executable-find "stack"))
    ...
```

In `flycheck-locate-sphink-source-directory`:
```
  (-when-let* ((filename (buffer-file-name))
               (dir (locate-dominating-file filename "conf.py")))
    (expand-file-name dir)))
```

In `flycheck-locate-config-file-ancestor-directories`:
```
(-when-let* ((basefile (buffer-file-name))
               (directory (locate-dominating-file basefile filename)))
    (expand-file-name filename directory)))
```

However, we can get rid of the `sphinx-source-directory` function by using the last one instead.